utils/slices.go

```diff
@@ -34,3 +37,27 @@ func All[T any](slice []T, f func(T) bool) bool {
 func AnyOf[T comparable](e T, values ...T) bool {
 	return slices.Contains(values, e)
 }
+
+// Unique returns a new slice with duplicates removed.
+// Panics if the slice contains pointer types.
+func Unique[T comparable](slice []T) []T {
```

Comment:
nit: Unique should be rename to Set as Set by definition doesn't have duplicates.